### PR TITLE
State: Refactor `chai` to `jest` in state tests

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

As we've been moving towards canonically using `jest` as our primary unit testing platform, we're slowly moving away from `chai`. 

This PR refactors the rest of the state tests to use `jest` instead of `chai`. That refactors more than half of the remaining `chai` usage. We also use the opportunity to refactor away from `sinon` in favor of the built-in jest mocking functionality where necessary.

Most of the legwork here has been done with codemods, but I had to do a few more additional changes manually due to the added complexity of the test helpers.

I'd appreciate an expedited review, since this touches a lot of files and might get stale quickly.

#### Testing instructions

Verify all tests pass: `yarn run test-client client/state`

### CODE REVIEW

File: client/state/help/ticket/test/actions.js [hunk 1]
```diff
@@ -72,17 +73,19 @@ describe( 'ticket-support/configuration actions', () => {
...
 			const action = ticketSupportConfigurationRequest()( spy );
 
-			assert( spy.calledWith( sinon.match( { type: HELP_TICKET_CONFIGURATION_REQUEST } ) ) );
+			expect(
+				spy.calledWith( sinon.match( { type: HELP_TICKET_CONFIGURATION_REQUEST } ) )
+			).toBeTruthy();
 
-			action.then( () => {
-				assert(
+			return action.then( () => {
```
Comment (noahtallen) [inline client/state/help/ticket/test/actions.js:80]: Why do we need to return the action promise in these cases?

**Review by noahtallen — APPROVED** [review]
I obviously couldn't check every line, but I scrolled through and looked at many examples, and everything looks pretty straightforward. I left a question about one thing I didn't understand, but I'm happy for this to be merged especially as all the unit tests are passing.

**Comment by stephanethomas** [comment]
I just had a look to changes for email providers (email forwards and G Suite), and they look good to me 👍.

File: client/state/help/ticket/test/actions.js (hunk 1 above)
Comment (tyxla) [reply client/state/help/ticket/test/actions.js:80]: Jest requires that we either return or await promises to be resolved when doing assertions on their result. See https://jestjs.io/docs/tutorial-async and [this particular ESLint rule](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/valid-expect-in-promise.md).

**Comment by tyxla** [comment]
Thank you both 🙌

**Pull request merged**